feeds/wifi-ax/hostapd/patches/900-hapd-netlink-ubus-bridge.patch

```diff
@@ -0,0 +1,891 @@
+ 	struct ubus_object obj;
+ 	struct avl_tree banned;
+ 	int notify_response;
++	struct avl_tree rt_events;
++};
++
++enum client_sess_type {
++	CST_ASSOC,
++	CST_AUTH,
++	CST_DISASSOC,
++	CST_FDATA,
++	CST_IP
++};
++
++struct client_assoc_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	char ssid[SSID_MAX_LEN];
++	int band;
++	int assoc_type;
++	int status;
++	int rssi;
++	int internal_sc;
++	uint8_t using11k;
++	uint8_t using11r;
++	uint8_t using11v;
++};
++
++struct client_disassoc_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	char ssid[SSID_MAX_LEN];
++	int band;
++	int assoc_type;
++	int status;
++	int rssi;
++	int internal_rc;
++};
++
++struct client_auth_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	char ssid[SSID_MAX_LEN];
++	int band;
++	uint32_t auth_status;
++};
++
++struct client_fdata_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	struct timespec rx_ts;
++	struct timespec tx_ts;
++};
++
++struct client_ip_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	unsigned char ip_addr[16];
++};
++
++struct client_session_record {
++	int type;
++	uint64_t timestamp;
++	union {
++		struct client_assoc_event assoc;
++		struct client_disassoc_event disassoc;
++		struct client_auth_event auth;
++		struct client_fdata_event fdata;
++		struct client_ip_event ip;
++	} u;
++};
++
++struct hostapd_event_avl_rec {
++	uint64_t session_id;
++	struct client_session_record *records;
++	size_t rec_nr;
++	struct avl_node avl;
+ };
+ 
+ void hostapd_ubus_add_iface(struct hostapd_iface *iface);
+@@ -45,6 +127,7 @@ void hostapd_ubus_add_bss(struct hostapd
+ void hostapd_ubus_free_bss(struct hostapd_data *hapd);
+ 
+ int hostapd_ubus_handle_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req);
++int hostapd_ubus_handle_rt_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req);
+ void hostapd_ubus_notify(struct hostapd_data *hapd, const char *type, const u8 *mac);
+ void hostapd_ubus_notify_beacon_report(struct hostapd_data *hapd,
+ 				       const u8 *addr, u8 token, u8 rep_mode,
+@@ -78,6 +161,11 @@ static inline int hostapd_ubus_handle_ev
+ {
+ 	return 0;
+ }
++
++static inline int hostapd_ubus_handle_rt_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req)
++{
++	return 0;
++}
+ 
+ static inline void hostapd_ubus_notify(struct hostapd_data *hapd, const char *type, const u8 *mac)
+ {
+--- a/src/ap/hostapd.c
++++ b/src/ap/hostapd.c
+@@ -1406,19 +1406,22 @@ static int hostapd_setup_bss(struct host
+ 				   "Generic snooping infrastructure initialization failed");
+ 			return -1;
+ 		}
+-
+-		if (dhcp_snoop_init(hapd)) {
+-			wpa_printf(MSG_ERROR,
+-				   "DHCP snooping initialization failed");
+-			return -1;
+-		}
+-
+ 		if (ndisc_snoop_init(hapd)) {
+ 			wpa_printf(MSG_ERROR,
+ 				   "Neighbor Discovery snooping initialization failed");
+ 			return -1;
+ 		}
+ 	}
++	if (dhcp_snoop_init(hapd)) {
```

Comment:
I thought you said there was an issue with the Client IP event.  This dhcp_snoop_init()  is part of that.  Are you sure you want this now?

Reply:
We should keep it and this would cause no harm because calling dhcp_snoop_init currently for Wifi6 would call the dummy version of it which does nothing and returns immediately. This is also the root cause of client IP event not getting generated in wifi6. More details under WIFI-1736